source/common/stats/statsd.cc

```diff
@@ -119,15 +119,21 @@ void TcpStatsdSink::TlsSink::commonFlush(const std::string& name, uint64_t value
   }
 
   // Produces something like "envoy.{}:{}|c\n"
-  memcpy(current_slice_mem_, StatPrefix, sizeof(StatPrefix) - 1);
-  current_slice_mem_ += sizeof(StatPrefix) - 1;
+  // This written this way for maximum perf since with a large number of stats and at a high flush
+  // rate this can become expensive.
+  const char* snapped_current = current_slice_mem_;
+  memcpy(current_slice_mem_, STAT_PREFIX, sizeof(STAT_PREFIX) - 1);
+  current_slice_mem_ += sizeof(STAT_PREFIX) - 1;
   memcpy(current_slice_mem_, name.c_str(), name.size());
   current_slice_mem_ += name.size();
   *current_slice_mem_++ = ':';
   current_slice_mem_ += StringUtil::itoa(current_slice_mem_, 30, value);
   *current_slice_mem_++ = '|';
   *current_slice_mem_++ = stat_type;
   *current_slice_mem_++ = '\n';
+
+  ASSERT(static_cast<uint64_t>(current_slice_mem_ - snapped_current) < name.size() + 40);
```

Comment:
Nit: Could make `name.size() + 40` a `const uint32_t` above and reference it in both locations with that.